### PR TITLE
Remove EKN_VERSION

### PR DESCRIPTION
**Pull request by ptomato**
https://phabricator.endlessm.com/T22395

### CODE REVIEW

File: dmodel/dm-utils.c [hunk 1]
```diff
@@ -299,30 +299,17 @@ dm_utils_id_get_hash (const char *id)
...
-
-  g_autoptr(GFile) ekn_version_file = g_file_get_child (dir, "EKN_VERSION");
-  g_autofree gchar *contents = NULL;
-  if (!g_file_load_contents (ekn_version_file, cancellable, &contents,
-                             NULL, NULL, error))
-    return NULL;
-
-  gchar *stripped = g_strstrip (contents);
-  return g_strdup (stripped);
+  return g_strdup("3");
```
Comment (cosimoc) [inline dmodel/dm-utils.c]: Needs space before paren

**Review by cosimoc — DISMISSED** [review]
This looks fine rather than a minor comment below, but why do you need to deprecate this instead of simply removing it? libdmodel is an entirely new library and this is the first release, so we can just drop it no?

**Comment by ebassi** [comment]
Like @cosimoc said, you cannot have a deprecation for something that hasn't been released, yet; having a "deprecated in 0.0" macro is literally an impossibility.

This is kind of what I was saying during the initial review of libdmodel: the first release of an API cycle should only have a "this is a public symbol" annotation — either something like `NAMESPACE_AVAILABLE_IN_1_0` or `NAMESPACE_AVAILABLE_IN_ALL`. You cannot have deprecations introduced in the first release, and you cannot mark a symbol not available until the first release, because there's literally no history prior to that point.

**Comment by ptomato** [comment]
OK, so we should bump the version to 0.1, make the symbol `DM_DEPRECATED_IN_0_2` and make a 0.2 stable release in SDK 4.1 in June?

I don't think we can remove the symbol, since we've technically "released" version 0.0 in SDK 4.0 today.

**Comment by ebassi** [comment]
The deprecation approach is fine, now that we've released libdmodel 0.0.

**Pull request merged**